src/raftstore/store/peer.rs

```diff
@@ -211,6 +211,8 @@ pub struct Peer {
 
     apply_scheduler: Scheduler<ApplyTask>,
 
+    pub pending_remove: bool,
```

Comment:
Yes. It's just like TiKV crash before it can write the tombstone key to rocksdb. In that situation, the peer will receive a gc message or greater ID message that will trigger the removal process again.